packages/vscode-extension/src/extension.ts

```diff
@@ -64,6 +71,25 @@ export async function activate(context: ExtensionContext) {
   // Handle any sequence events for activation
   await reconnectDatabaseConnectionOnExtensionActivation();
   await sendParametersToLanguageServer();
+
+  // in developement mode, we manually reload the extension.
+  if (process.env.watch === 'true') {
+    const watcher = workspace.createFileSystemWatcher(
+      new RelativePattern(
+        Uri.file(context.asAbsolutePath('dist')),
+        'extension.js',
+      ),
+    );
+
+    watcher.onDidChange(() => {
+      void window.showInformationMessage('Extension rebuilt, reloading...');
+      void commands.executeCommand<void>(
+        'workbench.action.restartExtensionHost',
+      );
+    });
+
+    context.subscriptions.push(watcher);
+  }
```

Comment:
All the docs I've found state there's no built in / clean way to do this. I'm open to removing it and requireing the dev to manually refresh the extensions instead if you prefer